launcher/java/JavaUtils.cpp

```diff
@@ -237,22 +231,25 @@ QList<JavaInstallPtr> JavaUtils::FindJavaFromRegistryKey(DWORD keyType, QString
                                       KEY_READ | KEY_WOW64_64KEY, &newKey) == ERROR_SUCCESS)
                     {
                         // Read the JavaHome value to find where Java is installed.
-                        value = new char[0];
+                        delete [] value;
+                        value = new TCHAR[0];
                         valueSz = 0;
                         if (RegQueryValueExW(newKey, keyJavaDir.toStdWString().c_str(), NULL, NULL, (BYTE *)value,
                                              &valueSz) == ERROR_MORE_DATA)
```

Comment:
What's the purpose in passing it an empty buffer? Just pass it NULL instead and like the documentation says, it will return ERROR_SUCCESS and output the size of the data in **bytes** into valueSz.
```suggestion
                        if (RegQueryValueExW(newKey, keyJavaDir.toStdWString().c_str(), NULL, NULL, NULL,
                                             &valueSz) == ERROR_SUCCESS && valueSz != 0)
```